### PR TITLE
Adding test for GPHDFS privileges

### PR DESCRIPTION
**Pull request by dgkimura**
GPHDFS has been removed in GPDB 6 and replaced with PXF. In order to
upgrade from 5 to 6 customer must drop the gphdfs privileges.

Co-authored-by: Bhuvnesh Chaudhary <bhuvnesh2703@gmail.com>

### CODE REVIEW

File: contrib/pg_upgrade/test/integration/output/gphdfs.source [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+
+-- gphdfs has been removed in GPDB 6 and replaced with PXF. In order to upgrade
+-- from 5 to 6 customer must drop the gphdfs privileges.
+ALTER ROLE gphdfsuser NOCREATEEXTTABLE(protocol='gphdfs', type='readable') NOCREATEEXTTABLE(protocol='gphdfs', type='writable');
+ALTER
+
+! ./gpdb6/bin/pg_upgrade --old-gp-dbid=1 --new-gp-dbid=1 --check --old-bindir=/home/dkimura/gpdb6/contrib/pg_upgrade/test/integration/gpdb5/bin --new-bindir=/home/dkimura/gpdb6/contrib/pg_upgrade/test/integration/gpdb6/bin --old-datadir=/home/dkimura/gpdb6/contrib/pg_upgrade/test/integration/gpdb5-data/qddir/demoDataDir-1 --new-datadir=/home/dkimura/gpdb6/contrib/pg_upgrade/test/integration/gpdb6-data/qddir/demoDataDir-1 --old-port=50000;
+
+*failure*
```
Comment (berlin-ab) [inline contrib/pg_upgrade/test/integration/output/gphdfs.source]: Do we intend for there to be a failure here, even after the `alter role`

File: contrib/pg_upgrade/test/integration/sql/upgraded_gphdfs.sql [hunk 2]
```diff
@@ -0,0 +1 @@
+SELECT rolname, rolsuper, rolinherit, rolcreaterole, rolcreatedb, rolcatupdate, rolcanlogin, rolreplication, rolconnlimit, rolpassword, rolvaliduntil, rolconfig, rolcreaterextgpfd, rolcreaterexthttp, rolcreatewextgpfd FROM pg_roles WHERE rolname='gphdfsuser';
```
Comment (berlin-ab) [inline contrib/pg_upgrade/test/integration/sql/upgraded_gphdfs.sql]: do we need all of these fields to verify what we're looking for? Is there a smaller subset - is true/false/count(*) sufficient?

File: contrib/pg_upgrade/test/integration/output/gphdfs.source (hunk 1 above)
Comment (dgkimura) [reply contrib/pg_upgrade/test/integration/output/gphdfs.source]: Ah, not really, thanks for catching.  I think the issue is that we're calling upgrade when the cluster is still up.  We could issue gpstop then run the check, but the check is actually redundant so we'll probably get rid of it.  We get it for free when we're upgrading the cluster as a whole.

File: contrib/pg_upgrade/test/integration/output/gphdfs.source [hunk 3]
```diff
@@ -0,0 +1,36 @@
+CREATE ROLE gphdfsuser WITH CREATEEXTTABLE(protocol='gphdfs', type='readable') CREATEEXTTABLE(protocol='gphdfs', type='writable');
+CREATE
+
+! ./gpdb6/bin/pg_upgrade --old-gp-dbid=1 --new-gp-dbid=1 --check --old-bindir=/home/dkimura/gpdb6/contrib/pg_upgrade/test/integration/gpdb5/bin --new-bindir=/home/dkimura/gpdb6/contrib/pg_upgrade/test/integration/gpdb6/bin --old-datadir=/home/dkimura/gpdb6/contrib/pg_upgrade/test/integration/gpdb5-data/qddir/demoDataDir-1 --new-datadir=/home/dkimura/gpdb6/contrib/pg_upgrade/test/integration/gpdb6-data/qddir/demoDataDir-1 --old-port=50000;
```
Comment (dgkimura) [inline contrib/pg_upgrade/test/integration/output/gphdfs.source]: Ugh... need to substitute the tokens `@ upgrade_test_path@`

File: contrib/pg_upgrade/test/integration/sql/upgraded_gphdfs.sql (hunk 2 above)
Comment (dgkimura) [reply contrib/pg_upgrade/test/integration/sql/upgraded_gphdfs.sql]: Probably not.  Actually the column we're most interested in is verifying that `rolcreaterexthdfs` doesn't exist.

**Pull request merged**